### PR TITLE
User Management

### PR DESCRIPTION
**Pull request by jamesmorgan**
* [x] Adding site disposition entity audit logs and cleaning dead entries
* [ ] Ability to Edit Currently Logged in User Details
 * [x] Allow edit of user name with persistence to `/users`
 * [ ] ~~Allow user picture upload~~ @andygray says don't bother with this. We can drop these into firebase directly!!
  * https://github.com/firebase/firepano
  * https://gist.github.com/abeisgreat/7330325
 * [ ]  Investigate intersection util - http://stackoverflow.com/questions/21500946/firebase-how-to-list-user-specific-data
* ~~[ ] Investigate Supplier -> Site relationships~~ - This can be on a different PR
* [x] Ready for review

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -202,6 +202,13 @@
 <script src="js/entity/site-disposition/modal/_module.js"></script>
 <script src="js/entity/site-disposition/modal/SiteDispositionModal.js"></script>
 
+<script src="js/entity/site-supplier/_module.js"></script>
+
+<script src="js/entity/user/_module.js"></script>
+<script src="js/entity/user/UserService.js"></script>
+<script src="js/entity/user/modal/_module.js"></script>
+<script src="js/entity/user/modal/EditUserModal.js"></script>
+
```
Comment (andygray) [inline app/index.html]: I must gulp this...I promised I seem to remember

File: app/js/entity/user/modal/EditUserModal.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        .service('EditUserModalService', EditUserModalService);
+
+    /**
+     * @ngInject
+     */
+    var EditUserModalCtrl = function ($scope, $modalInstance, user, UserService) {
+
+        var vm = this;
+
+        this.saveUser = function (user) {
```
Comment (andygray) [inline app/js/entity/user/modal/EditUserModal.js]: I A5 I did this slightly different. The modal closed passing the result back that could then be handled. This leaves the modal hanging for the promise to return?

File: app/index.html (hunk 1 above)
Comment (jamesmorgan) [inline app/index.html]: Yep we need it, also add ng-tempalcecahe as well to speed it up a bit.

File: app/js/entity/user/modal/EditUserModal.js (hunk 2 above)
Comment (jamesmorgan) [inline app/js/entity/user/modal/EditUserModal.js]: Yep I can change

File: app/index.html (hunk 1 above)
Comment (andygray) [inline app/index.html]: OK. Once I do a bit of dev, I will try gulping this up

File: app/index.html [hunk 3]
```diff
@@ -78,7 +80,9 @@
                     <img src="img/avatar3.png" class="img-circle" alt="User Image"/>
                 </div>
                 <div class="pull-left info">
-                    <p>Hello, {{ sidec.user.email }} [{{sidec.user.uid}}]</p>
+                    <p>Hello, {{ sidec.userData.name }}</p>
+
+                    <p>[{{sidec.userData.userId}}]</p>
```
Comment (jamesmorgan) [inline app/index.html]: This should be `uid`

**Comment by andygray** [comment]
Tested. Nice stuff. Don't need the image stuff.....yet :+1: :fire:

**Pull request merged**